### DISCUSSION TITLE
Does old getServerSideProps still available in Next 13v ? How to fetch from Firstore in Next 13 SSR ?

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#44115] [Help] [answered]
I am new to next 13 and When I use async function component for new ssr and I got error in npm run dev .

> Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.

![carbon (5)](https://user-images.githubusercontent.com/71011043/208279699-07cf3847-9e25-4b56-8f42-607cebb25861.png)

### **folder structure and nextjs v 13.0.6**
![Screenshot (116)](https://user-images.githubusercontent.com/71011043/208286099-a2482198-cf04-4288-8310-24f36a328bd5.png)

**Comment by Gbuomprisco** [comment, marked as answer]
You will need to use the Firebase Admin SDK when using it on the server; the web SDK will only work in browsers. As per Next.js, you shouldn't use hooks, that won't work.

**Reply by muhammadjufry** [reply]
Yes I've had the same issue I fixed that by adding next js fetching [swr ](https://swr.vercel.app/docs/with-nextjs)
Hope this work for you!

**Comment by thanhtutzaw** [comment]
Thanks you . I also learnt that new next 13 , new SSR is need server component .
